ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/QueryProjectNode.java

```diff
@@ -100,7 +100,7 @@ public QueryProjectNode(
     this.addAdditionalColumnsToIntermediateSchema = shouldAddAdditionalColumnsInSchema();
     this.outputSchema = buildOutputSchema(metaStore);
     this.intermediateSchema = QueryLogicalPlanUtil.buildIntermediateSchema(
-          source.getSchema(),
+          source.getSchema().withoutPseudoAndKeyColsInValue(),
```

Comment:
What's the purpose of this change? Is this fixing an existing bug, or something introduced with the KLIP-50 work?

Reply:
The purpose of this change is so that when there's no WHERE clause, which already does this call (In `QueryFilterNode`), we don't automatically include pseudo columns in our intermediate schema. I guess you could say this was a preexisting bug, but I don't think it was possible to surface it before the new pseudo columns because nothing unexpected existed in the source schema. The issue is that we build up the intermediate row which must agree with the intermediate schema, including the pseudo columns, but pull queries have no notion of ROWOFFSET or ROWPARTITION. So rather than trying to add support for it in the intermediate row, I've removed it from the schema.

Reply:
Got it; thanks for explaining!